ubcpi/ubcpi.py

```diff
@@ -0,0 +1,542 @@
+"""A Peer Instruction tool for edX by the University of British Columbia."""
+import os
+import random
+from copy import deepcopy
+import uuid
+
+from django.core.exceptions import PermissionDenied
+import pkg_resources
+from webob import Response
+from xblock.core import XBlock
+from xblock.exceptions import JsonHandlerError
+from xblock.fields import Scope, String, List, Dict, Integer
+from xblock.fragment import Fragment
+
+from answer_pool import offer_answer, validate_seeded_answers, get_other_answers
+import persistence as sas_api
+from serialize import parse_from_xml
+
+STATUS_NEW = 0
+STATUS_ANSWERED = 1
+STATUS_REVISED = 2
+
+# this is a limitation in the database submissions_submission.raw_answer field, which is a TEXT and has limit
+# of 64k in size. Because we are storing rationale and revised rationale both in the the field, the max size
+# for the rationale is half
+MAX_RATIONALE_SIZE = 32000
```

Comment:
Just let me clarify: we are not storing student rationale in the xblock storage at all. For those 50 answer candidate in the pool, we are just storing the student ID (32 bytes length?) which can be used to retrieve the rationale through submission API. So max pool size could be 50 * 32 bytes * number of options + some json overhead ({}, comma and column)). So for 4 options question, we are looking at 7k bytes storage. Here is then answer pool format for reference: https://github.com/ubc/ubcpi/blob/master/ubcpi/answer_pool.py#L18